modules/composer-workflowoperation/src/main/java/org/opencastproject/workflow/handler/composer/CutMarksToSmilWorkflowOperationHandler.java

```diff
@@ -0,0 +1,326 @@
+  }
+
+  /**
+   * The SMIL service to modify SMIL files.
+   */
+  private SmilService smilService;
+  public void setSmilService(SmilService smilService) {
+    this.smilService = smilService;
+  }
+
+  /** JSON Parser */
+  private static final Gson gson = new Gson();
+
+  /** Stores information read from JSON */
+  class Times {
+    private Long begin;
+    private Long duration;
+
+    public Long getStartTime() {
+      return begin;
+    }
+    public void setStartTime(Long startTime) {
+      this.begin = startTime;
+    }
+    public Long getDuration() {
+      return duration;
+    }
+    public void setDuration(Long duration) {
+      this.duration = duration;
+    }
+  }
+
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.workflow.api.WorkflowOperationHandler#start(org.opencastproject.workflow.api.WorkflowInstance,
+   *      JobContext)
+   */
+  @Override
+  public WorkflowOperationResult start(final WorkflowInstance workflowInstance, JobContext context)
+          throws WorkflowOperationException {
+    logger.info("Running cut marks to smil workflow operation on workflow {}", workflowInstance.getId());
+
+    List<MediaPackageElement> elementsToClean = new ArrayList<MediaPackageElement>();
+
+    try {
+      return cutMarksToSmil(workflowInstance.getMediaPackage(), workflowInstance.getCurrentOperation(), elementsToClean);
+    } catch (Exception e) {
+      throw new WorkflowOperationException(e);
+    } finally {
+      for (MediaPackageElement elem : elementsToClean) {
+        try {
+          workspace.delete(elem.getURI());
+        } catch (Exception e) {
+          logger.warn("Unable to delete element {}: {}", elem, e);
+        }
+      }
+    }
+  }
+
+  private WorkflowOperationResult cutMarksToSmil(MediaPackage src, WorkflowOperationInstance operation,
+          List<MediaPackageElement> elementsToClean) throws EncoderException, IOException, NotFoundException,
+          MediaPackageException, WorkflowOperationException, ServiceRegistryException {
+    final MediaPackage mediaPackage = (MediaPackage) src.clone();
+
+    // Read config options
+    final MediaPackageElementFlavor jsonFlavor = MediaPackageElementFlavor.parseFlavor(
+            getConfig(operation, SOURCE_JSON_FLAVOR));
+    final MediaPackageElementFlavor presenterFlavor = parseTargetFlavor(
+            getConfig(operation, SOURCE_PRESENTER_FLAVOR), "presenter");
+    final MediaPackageElementFlavor presentationFlavor = parseTargetFlavor(
+            getConfig(operation, SOURCE_PRESENTATION_FLAVOR), "presentation");
+    final MediaPackageElementFlavor targetSmilFlavor = MediaPackageElementFlavor.parseFlavor(
+            getConfig(operation, TARGET_SMIL_FLAVOR));
+
+    // Is there a catalog?
+    Catalog[] catalogs = mediaPackage.getCatalogs(jsonFlavor);
+    if (catalogs.length != 1) {
+      logger.warn("Number of catalogs in the source flavor does not equal one. Skipping...");
+      return skip(mediaPackage);
```

Comment:
The choice to skip instead of fail if no (or too many) catalogs are provided was made to be consistent with the behaviour of the VideoEditor WHO, which also does not fail if no cataloge is provided. Therefore I'd rather not fail here, although maybe it would be better to fail if there >1 cataloges.